ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py

```diff
@@ -9,41 +9,91 @@
 from ivy_tests.test_ivy.helpers import handle_cmd_line_args
 
 
-# prod
+@st.composite
+def _dtype_x_axis(draw, **kwargs):
+    dtype, x, shape = draw(helpers.dtype_and_values(**kwargs, ret_shape=True))
+    axis = draw(helpers.ints(min_value=0, max_value=len(shape) - 1))
+    if random.randint(0, 9) % 2 != 0:
```

Comment:
we should not use `random.randint` to generate randomness, all randonmess should come from hypothesis, something like `axis = draw(helpers.ints(min_value=0, max_value=len(shape) - 1) | st.just(None))` would be better. This might not run, but something along these lines will work.